### PR TITLE
Update stats log, count pending in foreign currencies, bumped Buxfer client and supported capabilities

### PR DESCRIPTION
**Pull request by nisan-tagar**
- Update the stats log format to support pending in foreign currency and updated transactions
- Bumped Buxfer client version and added support for automatic transaction status updates (from pending to cleared).
-  Buxfer store to report updates, and pending transactions in foreign currency.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -30,18 +30,21 @@ export type CategoryDef = {
 };
 
 export interface SaveStats {
-  name: string;
+  name: string; // Store name
```
Comment (daniel-hauser) [inline src/types.ts]: Using this kind of comment will show the text also in the IDE

```suggestion
  /**
   * The store name
   */
  name: string;
```

[also for the others]

File: src/types.ts [hunk 2]
```diff
@@ -30,18 +30,21 @@ export type CategoryDef = {
...
-  pending: number;
-  skipped: number;
-  existing: number;
+  total: number; // Total scrapped transactions handled
+  added: number; // Newly added to store
+  updated?: number; // Existing that changed and updated
+  pending: number; // Total scrapped transactions that are pending
+  skipped: number; // Transactions not added due to validation checks or the already exist
+  existing: number; // Scrapped transactions that already exists in store
+  foreign?: number; // Scrapped transactions that are charged in foreign currency (not ILS)
```
Comment (daniel-hauser) [inline src/types.ts]: Can you include the `foreign` field in the other storages as well?

File: src/storage/buxfer.ts [hunk 3]
```diff
@@ -118,4 +146,14 @@ export class BuxferStorage implements TransactionStorage {
...
     };
   }
+
+  logStats(stats: SaveStats): string {
+    return `
+    📝 ${stats.name} (${stats.table})
+    \t${stats.added} added
+    \t${stats.updated} updated
+    \t${stats.pending} pending
+    \t${stats.skipped} skipped (${stats.existing} existing ${stats.foreign} pending in foreign currency)
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts]: I prefer not to have a custom stats by storage and have one generic message.

Between the following options, do you find one of them useful?
1. Have `foreign` as a separate row (like the other stats) and Buxfer users will need to know they are skipped
2. Add a `otherCounters` object to SaveStats and send `otherCounters: { foreignSkipped: 100 }`

File: src/types.ts (hunk 1 above)
Comment (nisan-tagar) [reply src/types.ts]: Nice! Done.

File: src/types.ts (hunk 2 above)
Comment (nisan-tagar) [reply src/types.ts]: Sure, done. 
Updated UTs to use the required stats field

File: src/storage/buxfer.ts (hunk 3 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Sure.
I reverted the store interface in favor of the generic method approach. Lets go with option 1.
Looks like the changed stats format is failing the UTs "toMatchSnapshot()" assertions, is this resolved in the CI pipe when approved?

**Comment by nisan-tagar** [comment]
@daniel-hauser All change requests are now resolved, waiting for review. Thanks!

File: src/storage/buxfer.ts (hunk 3 above)
Comment (daniel-hauser) [reply src/storage/buxfer.ts]: > Looks like the changed stats format is failing the UTs "toMatchSnapshot()" assertions, is this resolved in the CI pipe when approved?

Snapshots souled be updated with `npm run test -- -u` and then be committed

**Comment by daniel-hauser** [comment]
@nisan-tagar I looked at the save stat logic and i see that the code is duplicated in all storages so i refactored to have all logic and creation in one place in #341
In the same PR i added the number of completed foreign transaction to the main message instead of adding them to the stats.

File: src/utils/currency.ts [hunk 4]
```diff
@@ -8,3 +10,19 @@ export function normalizeCurrency(currency: string | undefined) {
...
+ * @returns True if tx resolved in ILS currency
+ */
+export function checkIlsCharge(tx: TransactionRow): boolean {
+  return (
+    (tx.chargedCurrency != undefined &&
+      ILS_CURRENCY_LABELS.includes(tx.chargedCurrency.trim())) ||
+    (tx.originalCurrency != undefined &&
+      ILS_CURRENCY_LABELS.includes(tx.originalCurrency.trim()))
+  );
+}
```
Comment (daniel-hauser) [inline src/utils/currency.ts]:
```suggestion
/**
 * Foreign currencies resolve to ILS when settled after changing from pending status
 * @param tx Transaction checked
 * @returns True if tx resolved in ILS currency
 */
export function isIlsTransaction(tx: TransactionRow): boolean {
  return (
    normalizeCurrency(tx.chargedCurrency) === "ILS" ||
    normalizeCurrency(tx.originalCurrency) === "ILS"
  );
}
```

And if we must trim, let's do it inside `normalizeCurrency`

**Comment by nisan-tagar** [comment]
@daniel-hauser [341]( https://github.com/daniel-hauser/moneyman/pull/341) looks good.
I will close this PR, and join it to bump Buxfer 1.0.6

**Pull request closed without merging**